app/src/main/java/org/fossasia/openevent/events/ErrorDialogEvent.java

```diff
@@ -0,0 +1,59 @@
+package org.fossasia.openevent.events;
+
+import android.util.Log;
+
+import org.fossasia.openevent.OpenEventApp;
+import org.fossasia.openevent.api.APIClient;
+
+import retrofit.RetrofitError;
+
+/**
+ * Created by shivenmian on 28/12/15.
+ */
+public class ErrorDialogEvent {
+
+    public String ErrorType;
+    public String ErrorDesc;
+    public static final String TYPE = "RetrofitError Type";
+    public static final String ERROR_CODE = "Error Code";
+
+    public ErrorDialogEvent(String s1, String s2){
+        this.ErrorType = s1;
+        this.ErrorDesc = s2;
+    }
+
+    public void ErrorHandlerFunction(RetrofitError cause, ErrorDialogEvent event){
+
+        if (cause.getKind().equals(RetrofitError.Kind.CONVERSION)) {
```

Comment:
I've made it return an ErrorDialogEvent object.

I tried Switch case with cause.getKind() being compared, but just as I thought, it didn't work. 

When I took cases to be compared as RetrofitError.Kind.CONVERSION etc, it gave an error saying "Constant Expression required". And it's apt. RetrofitError.Kind.CONVERSION is not an object so the error came.

Then I took the cases as merely CONVERSION, NETWORK, HTTP etc. The code worked without errors. But then I took the base url as a malformed JSON ([this](http://www.mocky.io/v2/5682128c120000993893a2c1)) and ran. Instead of conversion error, it gave HTTP 200 error.

The if-else had worked perfectly with the malformed JSON and it did give me conversion error. The problem with switch is that it uses '==' and this compares objects as a whole, while .equals() also compares parts. That's why switch won't work in this case as I expected and saw. Check [this](http://stackoverflow.com/questions/1643067/whats-the-difference-between-equals-and).